### ISSUE TITLE
aws_lambda_function.vpc_config always detects a change when empty

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#443]
_This issue was originally opened by @rochdev as hashicorp/terraform#10900. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
0.8.2

### Affected Resource(s)
aws_lambda_function

### Terraform Configuration Files
```hcl
# Only relevant parts shown

variable "subnet_ids" {
  default = []
}

variable "security_group_ids" {
  default = []
}

resource "aws_lambda_function" "lambda" {
  vpc_config {
    subnet_ids         = ["${var.subnet_ids}"]
    security_group_ids = ["${var.security_group_ids}"]
  }
}
```

### Expected Behavior
Right after applying, `terraform plan` should not detect any change since there aren't any.

### Actual Behavior
Right after applying, `terraform plan` detects a change in `vpc_config`. If you apply the change, `terraform plan` will still detect the same change over and over again.

Relevant part of the output:

```text
-/+ module.auth.authorizer.authorizer_lambda.aws_lambda_function.lambda
    vpc_config.#:        "0" => "1" (forces new resource)
    vpc_config.0.vpc_id: "" => "<computed>"
```

### Steps to Reproduce
1. `terraform plan`
2. `terraform apply`
3. repeat

### References
Seem to be related:
- https://github.com/hashicorp/terraform/pull/6157
- https://github.com/hashicorp/terraform/pull/6191

**Comment by jgillard** [comment]
Still seeing this on 0.9.8

**Comment by Mandos** [comment]
I checked on 0.9.11 and it seems to be fixed.

**Comment by mikedal** [comment]
I am still seeing this issue in 0.9.11.
Like the original example, I am passing in subnet_ids and security_group_ids as variables, but I was also able to reproduce the bug like this
```HCL
vpc_config {
  security_group_ids = []
  subnet_ids         = []
}
```

**Comment by grahamlyons** [comment]
Still seeing this in Terraform 0.11.0 with AWS provider 1.5.0.

**Comment by nickm4062** [comment]
0.11.1 as well with 1.6.0

**Comment by ReallSlimmShady** [comment]
Still experiencing this issue on version v0.11.3

**Comment by bflad** [comment]
This might be resolved or at least partially fixed with something like #1341 and/or #3473

**Comment by oktaviandi-nugraha** [comment]
Anyone has any luck with this? I'm on version 0.11.5 and it's still happening. 

Ideally I'd like to create one module for both normal and in vpc lambda. If there's no workaround creating two modules seem the easiest workaround.

**Comment by jonwinton** [comment]
Still happening for me on 0.11.7 and AWS provider 1.20.0

**Comment by giuliocalzolari** [comment]
any ETA ?

**Comment by mdlavin** [comment]
@giuliocalzolari  If you are interested in testing out this feature in a patched v1.21.0 version, I've made some Alpine Linux x64 binaries available here: https://github.com/lifeomic/terraform-provider-aws/releases/tag/v1.21.0_patched_5f7d0def

That fork will likely see future patched versions as well. I haven't had a chance to update it for the latest release yet but I'm sure it will happen.

**Comment by saadismail** [comment]
I am still facing this on: 

> Terraform v0.11.7
> + provider.aws v1.29.0

I think this fix has not yet made to the public stable release.

**Comment by bflad** [comment]
In version 1.35.0 of the AWS provider, releasing shortly, the `aws_lambda_function` resource should now accept the following configuration and show no differences for an unconfigured VPC setup:

```hcl
vpc_config {
  security_group_ids = []
  subnet_ids         = []
}
```

**Comment by bflad** [comment]
This has been released in [version 1.35.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1350-september-06-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
